### ISSUE TITLE
tif image  decoding problem

### ISSUE THREAD

**Issue by dxw1997** [tensorflow/io#831]
I'm using tensorflow to build my model and the image format in my dataset is tif.
when I try to decode a tif-format image using tf.io.decode_image(), tensorflow tells me that it
doesn's support tif format.

To solve this problem, I use tf.py_function() to map a load function。But the dimensions of the loaded tensors are different , and my model can't  work.Can you help me with my tif-format image reading or the construction of the dataset?

This is my dataset building code:
`train_dataset_images = tf.data.Dataset.list_files(PATH+'/images/*.tif')`
`train_dataset_groundTruth = tf.data.Dataset.list_files(PATH+'/GT/*.tif')`
`train_dataset = tf.data.Dataset.zip((train_dataset_images, train_dataset_groundTruth))`
`train_dataset = train_dataset.map(lambda x,y:tf.py_function(func=load_image2, inp=[x,y], Tout=[tf.float32, tf.float32]))`
And This is the code of my load_image2 function and load function:
`def load_image2(image_file, gt_file):`
 ` input_image = load(image_file)`
  `input_groundTruth = load(gt_file)`
  #Resize input image to specified size
  `input_image = resize(input_image, IMAGE_HEIGHT, IMAGE_WIDTH)`
  `input_image = normalize(input_image)`
  `input_groundTruth = resize(input_groundTruth, IMAGE_HEIGHT, IMAGE_WIDTH)`
  `input_groundTruth = normalize(input_groundTruth)`
  `return input_image, input_groundTruth`

`def load(image_file:tf.Tensor):`

 ` image_file = image_file.numpy().decode('utf-8')`
  `img1 = Image.open(image_file)`
  
  `dtype = {'F': np.float32, 'L': np.uint8}[img1.mode]`
  `np_img1 = np.array(img1.getdata(), dtype=dtype)`
  `w, h = img1.size`
  `np_img1.shape = (h, w, np_img1.size // (w * h))`
  
  `#convert it to tensor`
  `t_img1 = tf.convert_to_tensor(np_img1) ` #(4441, 2206, 1)
  `input_image = tf.cast(t_img1, tf.float32)`
 
`  return input_image`

**Comment by yongtang** [comment]
@dxw1997 TIFF is supported in tensorflow-io's experimental API:

https://www.tensorflow.org/io/api_docs/python/tfio/experimental/image/decode_tiff

Note TIFF could have multiple images with different shape, so you have to provide an index (default 0) to specify which image inside the TIFF to use.

**Comment by kvignesh1420** [comment]
Closing as of now. Please reopen if required.
